snapshot-labs/sx-monorepo
# feat: add session keys (alias) for offchain spaces

### Summary



Closes: #282 

This is the last PR required to complete all the features needed in #282

This PR introduces `alias` (or session key), allowing a logged in user to send and save request on the sequencer, without being prompt for message signing.

This is done by signing the message via a proxy wallet (see #282 for description on how it works, or the original code from snapshot:

https://github.com/snapshot-labs/snapshot/blob/bcf4331b2496871eeb9ae4deca56562dba9287ad/src/composables/useAliasAction.ts#L1-L7

Alias signing is currently enabled on `follow` and `unfollow` spaces requests. 

Will be added to profile edition later

### How to test

1. Connect to an ETH wallet
3. Follow/Unfollow some spaces
4. On the first time, you will be prompt to sign a transaction to create an alias
5. After signing, the follow/unfollow request should succeed
6. Subsequent follow/unfollow requests should not ask for message signing again
7. Clear the localstorage for `ui.alias`
8. It should ask to sign an alias creation transaction again, when following/unfollowing a space.
9. Connect with a different wallet
10. It should not use the alias from previous wallet


NOTE: 

- You should not be prompted for signing a follow/unfollow tx anymore, only alias when not existing

### To-Do

- [x] Improve code
- [x] Waiting for #293 to be merged, and change this master base to `master`
- [ ] ~~Support alias for Starknet address~~ deferred to another PR
